lib/omniauth/strategies/azure_activedirectory.rb

```diff
@@ -254,7 +255,25 @@ def fetch_openid_config
       #
       # @return String
       def new_nonce
-        session['omniauth-azure-activedirectory.nonce'] = SecureRandom.uuid
+        session['omniauth-azure-activedirectory.nonce'] ||= []
```

Comment:
this would brake existing sessions where `session['omniauth-azure-activedirectory.nonce']` is String. Let's use `session['omniauth-azure-activedirectory.nonces']`?